File: app/src/main/java/com/github/alexeygorovoy/moxy_template/ui/demo/details/HeroDetailsPresenter.java
package com.github.alexeygorovoy.moxy_template.ui.demo.details;

import android.support.annotation.NonNull;

import com.arellomobile.mvp.InjectViewState;
import com.github.alexeygorovoy.moxy_template.api.models.Hero;
import com.github.alexeygorovoy.moxy_template.ui.common.moxy.BaseMvpPresenter;
import com.github.alexeygorovoy.moxy_template.ui.demo.details.view.HeroDetailsView;

@InjectViewState
public class HeroDetailsPresenter extends BaseMvpPresenter<HeroDetailsView> {

    private Hero hero;

    @Override
    protected void onFirstViewAttach() {
        super.onFirstViewAttach();
        getViewState().showHero(hero);
    }

    public void setHero(@NonNull Hero hero) {
        this.hero = hero;
    }
}

File: app/src/main/java/com/github/alexeygorovoy/moxytemplate/ui/demo/details/presenter/HeroDetailsPresenter.kt
package com.github.alexeygorovoy.moxytemplate.ui.demo.details.presenter

import com.arellomobile.mvp.InjectViewState
import com.github.alexeygorovoy.moxytemplate.api.models.Hero
import com.github.alexeygorovoy.moxytemplate.ui.common.moxy.BaseMvpPresenter
import com.github.alexeygorovoy.moxytemplate.ui.demo.details.view.HeroDetailsView
import javax.inject.Inject

@InjectViewState
class HeroDetailsPresenter @Inject internal constructor(
    private val hero: Hero
) : BaseMvpPresenter<HeroDetailsView>() {

    override fun onFirstViewAttach() {
        super.onFirstViewAttach()
        viewState.showHero(hero)
    }
}
